### ISSUE TITLE
Error: wrappedAction must be a tree node

### ISSUE THREAD

**Issue by Sladav** [xaviergonz/mobx-keystone#120]
I'm seeing the following error when trying provide an action as an event handler on a react component:

```
Uncaught Error: wrappedAction must be a tree node (usually a model or a shallow / deep child part of a model 'data' object)
```

It seems to be the difference in orphaning the function, ie: 
```
// this doesn't work
<Counter incrementCount={rootStore.incrementCount}/>

// but this does work
<Counter incrementCount={() => rootStore.incrementCount()}/>
```

It's pretty innocuous there, but I'm using a connect-style HOC and map the actions in an object and it feels a little less natural to wrap it in a lambda there...

```
const Counter = connect(PureCounter, (rootStore): Pick<PureCounterProps, InjectedKeys> => {
    return {
        count: rootStore.count,
        inc: () => rootStore.incrementCount(),
    };
});
```

This seems like a very minor issue, but I wanted to point it out because it never came up as an issue when I was using plain mobx or mobx-state-tree.

PS - Thanks for this library! We've been using using mobx without a lot of the bonus features offered with keystone and MST and (imo) ugly APIs and type issues with MST were always a hurdle for adopting it. I feel like this is the mobx-state-tree I've always wanted!

**Comment by xaviergonz** [comment]
That's an standard issue with react when using unbound methods, e.g:

```ts
class C {
  x = 10
  unbound() { this.x = 3 }
  bound = () => { this.x = 3 }
  lateBound() { this.x = 3 }
  constructor() {
    this.lateBound= this.lateBound.bind(this)
  }
}
const c = new C()

<Counter incrementCount={c.unbound}/> // "this" is not bound, so this.x is NOT accessible and won't work
<Counter incrementCount={c.bound}/> // "this" is bound, so this.x is accessible and will work
<Counter incrementCount={c.lateBound}/> // "this" is bound, so this.x is accessible and will work
```

From react docs:

https://reactjs.org/docs/handling-events.html

> You have to be careful about the meaning of this in JSX callbacks. In JavaScript, class methods are not bound by default. If you forget to bind this.handleClick and pass it to onClick, this will be undefined when the function is actually called.

> This is not React-specific behavior; it is a part of how functions work in JavaScript. Generally, if you refer to a method without () after it, such as onClick={this.handleClick}, you should bind that method.

This can be either fixed in user code:
```ts
@modelAction boundAction = () => { ... }
```
or in the library side, but I'd rather not fix it in the library side since unbound methods have a pro: since they reside in the prototype the same method memory/code can be shared (and optimized) across all instances of the class, where bound methods will generate a copy of the method per instance (and therefore the JVM needs to optimize them each time)

> PS - Thanks for this library! We've been using using mobx without a lot of the bonus features offered with keystone and MST and (imo) ugly APIs and type issues with MST were always a hurdle for adopting it. I feel like this is the mobx-state-tree I've always wanted!

Thanks! That's exactly what drove me to create mobx-keystone in the first place :) (I actually contributed quite a few stuff to MST prior to creating this lib)

**Comment by thehappycoder** [comment]
In my case,
```
@modelAction
onChange = (event: React.ChangeEvent<HTMLInputElement>): void = {}
```

still didn't work.

I assume mobx-keystone still wraps that inside a bindable function.

Ended up creating a callback in a functional component:
```
const onChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => form.onChange(event),
    [form]
  )
```

**Comment by ACampbell12** [comment]
Similar to @thehappycoder , altering my action to 
```js
@modelAction
onAuthChanged = (user: User): void = {
    ...
}
```
did not prevent this behavior when using this action with
```ts
onAttachedToRootStore() {
    APIService.subscribeToAuthChange(this.onAuthChanged);
}
```

I don't know enough about the internals of mobx-keystone to know if the following is ill-advised, but I was able to remove the error by manually binding the action:

```ts
onAttachedToRootStore() {
    this.onAuthChanged = this.onAuthChanged.bind(this);
    APIService.subscribeToAuthChange(this.onAuthChanged);
}
```

**Comment by xaviergonz** [comment]
I'll take a look

**Comment by xaviergonz** [comment]
Fixed in 0.69.4

**Comment by xaviergonz** [comment]
Btw, arrow functions were working, just not if you destructured them
